src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -54,13 +55,32 @@ public function getConfigTreeBuilder(): TreeBuilder
         $rootNode = $tb->getRootNode();
 
         $rootNode
+            ->beforeNormalization()
+                ->always()
+                ->then(function ($v) {
+                    if (isset($v['hide_user_not_found']) && !isset($v['expose_security_errors'])) {
+                        $v['expose_security_errors'] = $v['hide_user_not_found'] ? ExposeSecurityLevel::None : ExposeSecurityLevel::All;
+                    } elseif (!isset($v['expose_security_errors'])) {
+                        $v['expose_security_errors'] = ExposeSecurityLevel::None;
```

Comment:
This should **not** be the case. If a config does not configure the setting at all (neither through the old nor the new name), we should not force it to configure it as NONE, which would override a setting coming from a previous config file during merging.